src/ParamConverter/HashidsParamConverter.php

```diff
@@ -4,32 +4,33 @@
 
 namespace Roukmoute\HashidsBundle\ParamConverter;
 
-use Roukmoute\HashidsBundle\Hashids;
+use Hashids\HashidsInterface;
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
 use Sensio\Bundle\FrameworkExtraBundle\Request\ParamConverter\ParamConverterInterface;
 use Symfony\Component\HttpFoundation\Request;
 
 class HashidsParamConverter implements ParamConverterInterface
 {
-    /**
-     * @var Hashids
-     */
-    protected $hashids;
-    /**
-     * @var bool
-     */
-    private $passthrough;
-
-    public function __construct(Hashids $hashids, bool $passthrough)
+    private string $alphabet;
+    private bool $autoConvert;
+    private HashidsInterface $hashids;
+    private bool $passthrough;
+
+    public function __construct(HashidsInterface $hashids, bool $passthrough, bool $autoConvert, string $alphabet)
     {
         $this->hashids = $hashids;
         $this->passthrough = $passthrough;
+        $this->autoConvert = $autoConvert;
+        $this->alphabet = $alphabet;
     }
 
     public function apply(Request $request, ParamConverter $configuration): bool
     {
-        $this->setHashid($request, $configuration);
-        $this->removeHashidOption($configuration);
+        $decodedSuccessfuly = $this->decodeHashidOnRoute($request);
+
+        if ($decodedSuccessfuly === false) {
```

Comment:
I do not understand this condition.
Why can we not decode both an 'hashid' or 'id' and something from the configuration.

Reply:
It is kind an aliases.

Rework done.